Documentation should probably mention that Eigen includes should be added to search path

Since manif is based on Eigen, compiling manif programs also require finding Eigen. I recommend that something like the following be added to the README:

```cmake
find_package (Eigen3 REQUIRED)
target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC ${EIGEN3_INCLUDE_DIRS})
```

Yes, I did experience it. 

```jordi@chloe:~/tmp/manif/lol$ make
[ 50%] Building CXX object CMakeFiles/foo.dir/src/foo.o
In file included from /usr/local/include/manif/impl/lie_group_base.h:6,
                 from /usr/local/include/manif/SE2.h:5,
                 from /home/jordi/tmp/manif/lol/src/foo.cpp:104:
/usr/local/include/manif/impl/eigen.h:4:10: fatal error: Eigen/Core: No such file or directory
 #include <Eigen/Core>
          ^~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/foo.dir/build.make:63: CMakeFiles/foo.dir/src/foo.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:73: CMakeFiles/foo.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

I'm not sure what's going on, but the generated `manifTargets.cmake`file at `/usr/local/share/manif/cmake` does not have the proper directives for including Eigen 3 directories.

This is on Debian 10.1

--- a/README.md
+++ b/README.md
@@ -105,6 +105,9 @@ In your project `CMakeLists.txt` :
 
 ```cmake
 project(foo)
+# Find the Eigen library
+find_package(Eigen3 REQUIRED)
+target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC ${EIGEN3_INCLUDE_DIRS})
 # Find the manif library
 find_package(manif REQUIRED)
 add_executable(${PROJECT_NAME} src/foo.cpp)

